File: javascript/anagram/anagram.js
const wordLetters = (word) => word.toLowerCase().split('').sort().join('');

export const findAnagrams = (baseWord, words) => {
  return words.filter(word => {
    if (word.toLowerCase() === baseWord.toLowerCase() || word.length !== baseWord.length) {
      return false;
    }

    return wordLetters(baseWord) === wordLetters(word);
  });
}

File: python/anagram/anagram.py
def find_anagrams(word, candidates):
    word_lower = word.lower()
    word_letters = "".join(sorted(word_lower))

    def is_anagram(candidate):
        candidate_lower = candidate.lower()
        candidate_letters = ''.join(sorted(candidate_lower))

        return candidate_letters == word_letters and candidate_lower != word_lower

    return filter(is_anagram, candidates)
